src/v/raft/tests/raft_reconfiguration_test.cc

```diff
@@ -497,11 +497,69 @@ TEST_F_CORO(raft_fixture, test_force_reconfiguration) {
 
     std::vector<vnode> base_replica_set = all_vnodes();
     size_t reconfiguration_count = 0;
+    model::revision_id next_rev{1};
 
     auto current_replicas = base_replica_set;
 
     vlog(logger().info, "initial replicas: {}", current_replicas);
-
+    auto reconfigure_until_success =
+      [&](model::revision_id rev, raft::vnode to_skip) {
+          auto deadline = model::timeout_clock::now() + 30s;
+          return ss::repeat([&, this, rev, deadline] {
+              vassert(
+                model::timeout_clock::now() < deadline,
+                "Timeout waiting for reconfiguration");
+              return ss::parallel_for_each(
+                       nodes().begin(),
+                       nodes().end(),
+                       [&, to_skip, rev](const raft_nodes_t::value_type& pair) {
+                           auto raft = pair.second->raft();
+
+                           if (
+                             pair.second->get_vnode() == to_skip
+                             && !pair.second->raft()->is_leader()) {
```

Comment:
> && !pair.second->raft()->is_leader()

Unrelated question, this check is in place so that the same doesn't win the election again and keep the old configuration, right?